pkg/distributor/query.go

```diff
@@ -266,6 +266,8 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSets [
 				return ingesterQueryResult{}, err
 			}
 
+			defer resp.FreeBuffer()
```

Comment:
This is a `for` loop on the stream, and we're getting multiple responses here, and we're retaining a reference to each one of them until the end of the execution. We should free the buffer as soon as we've used it.

Reply:
Good catch @colega, I missed that. Fixed, PTAL.